NOTES.md

```diff
@@ -0,0 +1,84 @@
+## Notes, Tips + Tricks, and General Study Guide for Interviews
+
+### Table of Contents
+- [Definitions](#definitions)
+- [Tips + Tricks](#tips--tricks)
+
+### Definitions
+
+**Set**
+> A list of elements in any order. In general, sets are distinguished by the elements in them and not the specific ordering so sets with the same elements but a different ordering are considered duplicates. An empty set is considered to be a `null set` containing only the `null` element.
+
+Ex. [ 1, 2, 3, 4, 5 ]
+
+**Subset**
+> Given a set `a` of elements, a subset would be another set `b` containing only elements from `a`. 
+All sets contain the null element, therefore the null set is always a subset of any set. 
+Because set `b` can only contain elements from `a`, the size of `b` is at most the size of `a`. 
+For a set with `N` elements, there are exactly `2^N` subsets because for every element there are two choices: include the element or exclude the element. 
+This can be visualized as a binary tree when making the recursive call.
+
+Ex. `[ 1, 2, 3, 4, 5 ] => [ 2, 4 ]`
+
+**Permutations**
+> Given a set `a` of elements, a permutation of `a` would be a unique ordering of the elements in `a`. 
+Because the ordering matters for this, the length of a permutation is always equal to the length of the original set.
```

Comment:
@Puyush6889 if you want to wrap text in bold, then surround the text with two asterisks (with no spaces)

```
** this is bold **  (with spaces)
**this is bold** (without spaces)
```